gdal/swig/python/osgeo/utils/gdal_calc.py

```diff
@@ -129,57 +199,144 @@ def doit(opts, args):
     # set up global namespace for eval with all functions of gdalnumeric
     global_namespace = dict([(key, getattr(gdalnumeric, key))
                              for key in dir(gdalnumeric) if not key.startswith('__')])
+    if opts.user_namespace:
+        global_namespace.update(opts.user_namespace)
 
     if not opts.calc:
         raise Exception("No calculation provided.")
     elif not opts.outF:
-        raise Exception("No output file provided.")
+        if not opts.return_ds:
```

Comment:
do we really need that attribute ? Cannot we just always return the dataset ? People wanting to create a in-memory dataset can do that with an empty output dataset name (or whatever they want) and select the MEM driver

Reply:
sounds right.